### ISSUE TITLE
[A11y] Fix role relationship error in Annotations component

### ISSUE THREAD

**Issue by Dananji** [samvera-labs/ramp#811]
### Description

Error description from SiteImprove plugin;

> Role with implied hidden content has keyboard focus: Some elements have a role that implies their internal structure should be hidden from assistive technologies. In other words, their sub-elements have a presentational role.
If a sub-element has keyboard focus, it should be moved outside of the presentation role.

This is referring to the annotations with links in the text. In the current implementation, annotations component is displaying annotations as list elements with `role='option'`. And any children (links within annotation texts) of such an element are considered presentational.
Relevant success criterion: https://www.w3.org/TR/WCAG/#name-role-value (Level A)

### Done Looks Like
- [ ] Identify a suitable role for the annotations elements and replace the existing role with it
- [ ] Access the accessibility design pattern adapted for keyboard navigation is viable

### Note
This error is introduced in the work done in #781, which has an open PR for implementing keyboard navigation for the annotation component. If/when this work is merged to `main`, we will see this accessibility issue in the annotations component.

**Comment by Dananji** [comment]
Changes made to fix this issue:
- replace `<ul>/<li>` elements with `<div>` as the list related HTML elements implicitly add list-based role relationships to the DOM structure, which doesn't allow annotation rows to present links/buttons
- only the timestamp portion of the annotation is presented as a button. With this, the keyboard navigation will only focus on the timestamp portion when using `ArrowUp` and `ArrowDown` keys to traverse the annotation content
  - additionally, add `aria-label` attribute with time and text to this time-tag `<div>` element in the annotation row to be read by screen readers when focus is on the annotation timestamp. 

This work is waiting on #805 to be reviewed and merged.

**Comment by joncameron** [comment]
This is fixed and no longer reported in SiteImprove.
